tools/tuf/test/Orbit-auto-update-test-guide.md

```diff
@@ -0,0 +1,151 @@
+# Orbit auto-update test guide
+
+## Setup
+
+To test Orbit we must have a setup for the three OSs where we support Orbit: macOS, Linux and Windows.
+
+This guide assumes:
+- A macOS host OS, where we'll run most of the commands, TUF server, Orbit and the Fleet server.
+- Two VMWare VMs, with Windows 10 and Ubuntu 22.04, where we'll run Orbit.
+- The two guest OSs will connect to the host OS via the `host.docker.internal` hostname.
+To do this, you can add an entry like `192.168.103.1	host.docker.internal` to the `hosts` file in the VMs
+(`/etc/hosts` on Linux and `C:\Windows\System32\drivers\etc\hosts` on Windows).
+- The host OS can share packages with the guest OSs (via VMWare's shared folders feature).
+
+## Last release
+
+Head over to https://github.com/fleetdm/fleet/releases and grab the git tag of the last releases for Fleet and Orbit.
+
+At the time of writing:
+- Last Orbit release: `orbit-v1.2.0`
+- Last Fleet release: `fleet-v4.21.0`
+
+## Run Fleet
+
+```sh
+git checkout fleet-v4.21.0
+
+make fleet fleetctl
+make db-reset
+./build/fleet serve --logging_debug --dev --dev_license
+
+./build/fleetctl setup \
+    --email foo@example.com \
+    --name foo \
+    --password p4ssw0rd.123 \
+    --org-name "Fleet Device Management Inc."
+
+export ENROLL_SECRET=K3lOqio9XKw6Cr24qw1XyCRzydwRZeAv
+echo "---\napiVersion: v1\nkind: enroll_secret\nspec:\n  secrets:\n  - secret: $ENROLL_SECRET\n" > secrets.yml
+./build/fleetctl apply -f secrets.yml
+```
+
+## Generate local TUF repository
+
+1. The following commands will generate the TUF repository with the last released version of Orbit and automatically generate the Orbit packages.
+
+```sh
+git checkout orbit-v1.2.0
+rm -rf test_tuf
+
+SYSTEMS="macos windows linux" \
+PKG_FLEET_URL=https://localhost:8080 \
+PKG_TUF_URL=http://localhost:8081 \
+DEB_FLEET_URL=https://host.docker.internal:8080 \
+DEB_TUF_URL=http://host.docker.internal:8081 \
+RPM_FLEET_URL=https://host.docker.internal:8080 \
+RPM_TUF_URL=http://host.docker.internal:8081 \
+MSI_FLEET_URL=https://host.docker.internal:8080 \
+MSI_TUF_URL=http://host.docker.internal:8081 \
+GENERATE_PKG=1 \
+GENERATE_DEB=1 \
+GENERATE_RPM=1 \
+GENERATE_MSI=1 \
+ENROLL_SECRET=$ENROLL_SECRET \
+FLEET_DESKTOP=1 \
+FLEET_CERTIFICATE=1 \
+./tools/tuf/test/main.sh
+```
+
+2. Install Orbit on the host (macOS):
+```sh
+sudo installer -pkg fleet-osquery.pkg -verbose -target /
```

Comment:
It's very likely that you'll have to do this more than once (as you'll probably find a bug or two), should we add a note somewhere to make sure you're not running an old version? I generally run:

```
$ sudo rm -rf /var/log/orbit/orbit.stderr.log
$ orbit/tools/cleanup/cleanup_macos.sh
```